net/tcp/tcp_seqno.c

```diff
@@ -142,6 +144,42 @@ uint32_t tcp_addsequence(FAR uint8_t *seqno, uint16_t len)
 
 void tcp_initsequence(FAR uint8_t *seqno)
 {
+  int ret;
+
+  /* If g_tcpsequence is already initialized, just copy it */
+
+  if (g_tcpsequence != 0)
+    {
+      goto copy_seqno;
```

Comment:
nitpick: Maybe we can try avoid using `goto` and just
```
if (g_tcpsequence == 0)
  {
    // place all random logic here
  }

tcp_setsequence(seqno, g_tcpsequence);
```

Reply:
Yes, good idea! Fixed!